### PR TITLE
MS SQL and Oracle support

### PR DESCRIPTION
**Pull request by renatocaval**
Draft because on top of #263.
Ready for review otherwise.

The main reason I'm sending this apart is that it may have some legal/license implications. 

1. Oracle driver is not available in Maven Central. We need to download it and add to the `lib` folder. We won't distribute it, so I think it's fine but this need to be confirmed.
1. Microsoft docker image requires a license agreement. It works by adding a file under `src/test/resource`. Again, we won't be distributing it, but needs to be confirmed if that's ok. People checking out the projection will be building it without 'signing' it themselves. 

References #23

### CODE REVIEW

File: akka-projection-jdbc/src/main/scala/akka/projection/jdbc/internal/JdbcSettings.scala [hunk 1]
```diff
@@ -31,6 +31,7 @@ private[projection] case class JdbcSettings(config: Config, executionContext: Ex
 
     dialectToLoad match {
       case "mysql-dialect"   => MySQLDialect(schema, table)
+      case "mssql-dialect"   => MSSQLServerDialect(schema, table)
```
Comment (patriknw) [inline akka-projection-jdbc/src/main/scala/akka/projection/jdbc/internal/JdbcSettings.scala:34]: oracle also?

File: akka-projection-jdbc/src/test/resources/container-license-acceptance.txt [hunk 2]
```diff
@@ -0,0 +1 @@
+mcr.microsoft.com/mssql/server:2017-CU12
```
Comment (patriknw) [inline akka-projection-jdbc/src/test/resources/container-license-acceptance.txt:1]: we concluded a different approach

File: akka-projection-jdbc/src/main/scala/akka/projection/jdbc/internal/Dialect.scala [hunk 3]
```diff
@@ -162,3 +162,88 @@ private[jdbc] case class MySQLDialect(schema: Option[String], tableName: String)
...
+  override def insertStatement(): String = DialectDefaults.insertStatement(table)
+
+  override def updateStatement(): String = DialectDefaults.updateStatement(table)
+}
+
+/**
+ * INTERNAL API
+ */
+@InternalApi
+private[jdbc] case class OracleServerDialect(schema: Option[String], tableName: String) extends Dialect {
```
Comment (patriknw) [inline akka-projection-jdbc/src/main/scala/akka/projection/jdbc/internal/Dialect.scala:206]: you found oracle driver in maven central so lib/ojdbc8.jar can be removed

**Comment by renatocaval** [comment]
> looking good, but not fully updated yet I guess

In the meantime I had it updated locally. Needs clean up and will probably put Oracle on another PR because of the batch issues.

**Comment by renatocaval** [comment]
closing this one. Will sent in separate and on top of the documentation PR.

**Pull request closed without merging**